grabl/analysis/DependencyAnalysis.kt

```diff
@@ -0,0 +1,46 @@
+package grabl.analysis
+
+import com.eclipsesource.json.Json
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.util.regex.Pattern
+
+
+fun main() {
+    val remotePattern = Pattern.compile("(remote) = \"https://github.com/(.*)\"(?:.*)");
+    val refPattern = Pattern.compile("(commit|tag) = \"(.*)\"(?:.*)");
+
+    var workspaceDirectory = System.getenv("BUILD_WORKSPACE_DIRECTORY")
+    var dependencies = Paths.get(workspaceDirectory, "dependencies", "graknlabs", "dependencies.bzl")
+    var repositoriesArray = Json.array()
+
+
+    Files.readAllLines(dependencies).windowed(size=2, step=1).map { window ->
+        window.map { it.trim(' ', ',', '\t') }
+    }.mapNotNull { window ->
+        val remotePatternMatcher = remotePattern.matcher(window.first())
+        val refPatternMatcher = refPattern.matcher(window.last())
+        if (!(remotePatternMatcher.matches() && refPatternMatcher.matches())) {
+            return@mapNotNull null
+        }
+        mapOf(
+                remotePatternMatcher.group(1) to remotePatternMatcher.group(2),
```

Comment:
Is it possible to somehow name the group so that it's slightly more readable?

Reply:
@lriuui0x0 addressed in https://github.com/graknlabs/dependencies/pull/210/commits/38fc9d8aa6ac2fb6b7fe21a82cb216b9867c41ca